### PR TITLE
Various changes

### LINKED ISSUE #1492: Remove public facing `iter8` commands
Remove unused iter8 commands:

- [ ] `iter8 k launch` (replaced by `helm upgrade [--install]`)
- [ ] `iter8 k log` (replaced by `kubectl logs`)
- [ ] `iter8 k delete` (replaced by `helm delete`)

### LINKED ISSUE #1598: Bump go version for workflows
Workflows are currently referencing go version `1.19` however the latest version is `1.21`.

### LINKED ISSUE #1611: Remove duplicate payload templates for `http` and `slack`
The same payload templates for `http` and `slack` exists in both `charts/iter8` and `templates/notify`. We should remove one.

Generally leaning towards removing `charts/iter8`. In addition, we also need to verify the default `payloadTemplateURL` in both the `slack` and `github` tasks as well as in the docs.

### LINKED ISSUE #1617: Align iter8 to match terminology used in docs
With v0.16, a number of concepts have been removed, which have been eliminated in the docs but not in iter8

For example:

* The metric server expects an `experiment` query parameter for both the HTTP and gRPC dashboards, but the term `experiment` has been removed from the documentation. Some alternatives would be `test` or `performanceTest`.

  Query string: `namespace=default&experiment=default`

* The `slack` and `github` tasks (which under the covers is the `notify` task) both use a payload template that has access to a `.Report` field. The documentation makes no mention of this. The documentation makes mention of an experiment summary, however.

### PR DESCRIPTION
**Pull request by Alan-Cha**
Fix #1598
Fix #1615 
Fix #1611 
Fix #1617 
Fix #1492

### CODE REVIEW

File: cmd/k.go [hunk 1]
```diff
@@ -15,7 +15,8 @@ var kcmd = &cobra.Command{
 	Long:  "Work with Kubernetes experiments",
 }
 
-// addExperimentGroupFlag adds the experiment group flag
+// TODO: is group still needed?
+// addExperimentGroupFlag adds the test group flag
 func addExperimentGroupFlag(cmd *cobra.Command, groupP *string) {
 	cmd.Flags().StringVarP(groupP, "group", "g", driver.DefaultExperimentGroup, "name of the experiment group")
 }
```
Comment (Alan-Cha) [inline cmd/k.go]: Group is now the name of the experiment right? Does this need to be renamed?

File: cmd/krun.go [hunk 2]
```diff
@@ -23,7 +23,7 @@ func newKRunCmd(kd *driver.KubeDriver, out io.Writer) *cobra.Command {
 	actor.EnvSettings = settings
 	cmd := &cobra.Command{
 		Use:          "run",
-		Short:        "Run a Kubernetes experiment",
+		Short:        "Run a Kubernetes test",
```
Comment (Alan-Cha) [inline cmd/krun.go:26]: Revert these changes

File: cmd/k.go (hunk 1 above)
Comment (kalantar) [reply cmd/k.go]: Agree that this flag is no longer needed.

File: cmd/krun.go [hunk 3]
```diff
@@ -10,11 +10,11 @@ import (
 
 // krunDesc is the description of the k run command
 const krunDesc = `
-Run a Kubernetes experiment. This command reads an experiment specified in a secret and writes the result back to the secret.
+Run a Kubernetes test. This command reads an test specified in a secret and writes the result back to the secret.
```
Comment (kalantar) [inline cmd/krun.go]: - "Run a performance test on Kubernetes"
- "reads a test" (not an)

File: cmd/krun.go (hunk 2 above)
Comment (kalantar) [inline cmd/krun.go]: "Run a performance test"

File: charts/iter8/README.md [hunk 4]
```diff
@@ -1,3 +1,3 @@
-# Iter8 experiment chart
+# Iter8 test chart
```
Comment (kalantar) [inline charts/iter8/README.md]: perhaps this is a test "runner" chart?

File: cmd/krun.go [hunk 5]
```diff
@@ -10,11 +10,11 @@ import (
...
 
-	$ iter8 k run --namespace {{ .Experiment.Namespace }} --group {{ .Experiment.group }}
+	$ iter8 k run --namespace {{ .Test.Namespace }} --group {{ .Test.group }}
```
Comment (kalantar) [inline cmd/krun.go]: remove `--group`

File: cmd/root.go [hunk 6]
```diff
@@ -47,16 +45,6 @@ func Execute() {
 	cobra.CheckErr(rootCmd.Execute())
 }
```
Comment (kalantar) [inline cmd/root.go:47]: Isn't this still needed to support reading of value.yaml (and overriding it). for iter8 and controller?

File: cmd/krun.go (hunk 5 above)
Comment (kalantar) [reply cmd/krun.go]: also change _k-job.tpl to remove this option

File: cmd/k.go (hunk 1 above)
Comment (kalantar) [reply cmd/k.go]: Actually, how is the `--group` option used by `iter8 k run`? Do we still need it for that or can we get it from a name instead?  If not, it might still be needed.

File: cmd/root.go (hunk 6 above)
Comment (Alan-Cha) [reply cmd/root.go:47]: The linter said that this function was not used so I removed it but I think you are right

Comment (Alan-Cha) [reply cmd/root.go:47]: This function was only used in `iter8 k launch`, not in `iter8 k run`.

File: base/collect_http.go [hunk 7]
```diff
@@ -81,30 +82,6 @@ var (
...
-			return true
-		}
-		// if both limits are present (check both)
-		if lims.Upper != nil && lims.Lower != nil && code <= *lims.Upper && code >= *lims.Lower {
-			return true
-		}
-	}
-	return false
-}
-
```
Comment (Alan-Cha) [inline base/collect_http.go:107]: This is not used

**Review by kalantar — DISMISSED** [review]
looks good

**Review by kalantar — COMMENTED** [review]
lgtm
